src/hooks/gamebananaApi.ts

```diff
@@ -228,26 +228,136 @@ export const useGamebananaModImageUrls = (
 
 
 
-export const GAMEBANANA_OLYMPUS_ICON_URL = "https://images.gamebanana.com/img/ico/tools/60b506516b5dc.png";
+type GamebananaFileMetadata = {
+    "_idRow": string,
+    "_sFile": string,
+    "_nFilesize": number,
+    "_sDescription": string,
+    "_tsDateAdded": number,
+    "_nDownloadCount": number,
+    "_sAnalysisState": string,
+    "_sDownloadUrl": string,
+    "_sMd5Checksum": string,
+    "_sClamAvResult": string,
+    "_sAnalysisResult": string,
+    "_bContainsExe": boolean;
+};
+
+
+const isGamebananaFileMetadata = (
+    data: unknown
+): data is GamebananaFileMetadata => {
```

Comment:
I'm not familiar with this syntax, why is the `data is GamebananaFileMetadata` needed?

Reply:
It's a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates). It's there to explicitly tell TypeScript "If this function returns true, the variable that was passed into the `data` parameter is *definitely* of type `GamebananaFileMetadata`. That way, after line 337 TypeScript can infer the correct type for `filesObject`.